### PR TITLE
remove prettylog and split logging utils

### PR DESCRIPTION
**Pull request by mosquito**

### CODE REVIEW

File: aiomisc/log/formatter/json.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+        data = dict(errno=0 if not record.exc_info else 255)
+
+        for key, value in self.FIELD_MAPPING.items():
+            mapping, field_type = value
+
+            v = record_dict.get(key)
+
+            if not isinstance(v, field_type):
+                v = field_type(v)
```
Comment (dizballanze) [inline aiomisc/log/formatter/json.py:50]: Why we need this? It's pretty slow I guess

File: aiomisc/log/formatter/json.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            'level': self.LEVELS[record.levelno],
+        }
+
+        if self.datefmt:
+            payload['@timestamp'] = self.formatTime(record, self.datefmt)
+
+        if record.exc_info:
+            payload['stackTrace'] = self.formatException(record.exc_info)
+
+        return json.dumps(payload, ensure_ascii=False)
```
Comment (dizballanze) [inline aiomisc/log/formatter/json.py:82]: maybe we should add support for custom json serializer

**Review by dizballanze — CHANGES_REQUESTED** [review]

File: aiomisc/log/formatter/json.py (hunk 1 above)
Comment (mosquito) [reply aiomisc/log/formatter/json.py:50]: `isinstance` is not so heavy. And the fields mapping has not variable length.

File: aiomisc/log/formatter/json.py (hunk 2 above)
Comment (mosquito) [reply aiomisc/log/formatter/json.py:82]: ok.

**Review by alvassin — APPROVED** [review]

**Review by dizballanze — APPROVED** [review]

File: aiomisc/log/wrap.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        with suppress(Exception):
+            if handler.buffer:
+                handler.flush()
+
+        time.sleep(flush_interval)
+
+
+def wrap_logging_handler(
+    handler: logging.Handler,
+    loop: asyncio.AbstractEventLoop = None,
```
Comment (leenr) [inline aiomisc/log/wrap.py:37]: `Optional[asyncio.AbstractEventLoop]`

File: aiomisc/log/formatter/json.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+class JSONLogFormatter(logging.Formatter):
+    JSON_DUMPS = _dump_json
+
+    LEVELS = MappingProxyType({
+        logging.CRITICAL: "crit",
+        logging.FATAL: "fatal",
+        logging.ERROR: "error",
+        logging.WARNING: "warn",
+        logging.WARN: "warn",
```
Comment (leenr) [inline aiomisc/log/formatter/json.py:21]: It's value the same as `logging.WARNING`.
```python
>>> logging.WARN
30
>>> logging.WARNING
30
```

File: aiomisc/log/enum.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    @classmethod
+    def choices(cls):
+        return tuple(cls._member_names_)
+
+
+class DateFormat(Enum):
+    color = '%Y-%m-%d %H:%M:%S'
+    stream = '[%Y-%m-%d %H:%M:%S]'
+    # Optimization: float value will be returned
```
Comment (leenr) [inline aiomisc/log/enum.py:33]: Where?

File: aiomisc/log/config.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+from .enum import LogFormat
+from .formatter import json_formatter, color_formatter
+from .wrap import wrap_logging_handler
+
+DEFAULT_FORMAT = '%(levelname)s:%(name)s:%(message)s'
+
+
+def create_logging_handler(log_format: LogFormat = LogFormat.color,
+                           date_format=None, **kwargs):
```
Comment (leenr) [inline aiomisc/log/config.py:14]: Missing type annotation:
```python
date_format: Optional[LogFormat]
```

File: aiomisc/log/config.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+def basic_config(
+    level: int = logging.INFO,
+    log_format: Union[str, LogFormat] = LogFormat.color,
+    buffered: bool = True, buffer_size: int = 1024,
+    flush_interval: float = 0.2, loop=None, **kwargs
+):
+
+    if isinstance(level, str):
+        level = LogFormat[level]
```
Comment (leenr) [inline aiomisc/log/config.py:57]: Why `LogFormat` here and not `LogLevel`?

File: aiomisc/log/wrap.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    buffer_size: int = 1024,
+    flush_interval: float = 0.1,
```
Comment (leenr) [inline aiomisc/log/wrap.py:39]: `Union[float, int]`? `_thread_flusher` has support for both.

File: aiomisc/log/config.py [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+from typing import Union
+
+from .enum import LogFormat
+from .formatter import json_formatter, color_formatter
+from .wrap import wrap_logging_handler
+
+DEFAULT_FORMAT = '%(levelname)s:%(name)s:%(message)s'
+
+
+def create_logging_handler(log_format: LogFormat = LogFormat.color,
```
Comment (leenr) [inline aiomisc/log/config.py:13]: Why we don't allow users to pass their own formatter?
Especially when functions and classes inside `aiomisc.log.formatter` has some degree of customization.

**Review by leenr — CHANGES_REQUESTED** [review]

File: aiomisc/log/config.py (hunk 9 above)
Comment (mosquito) [reply aiomisc/log/config.py:13]: This PR just about moving the code and doesn't brings a new features.

File: aiomisc/log/config.py (hunk 7 above)
Comment (mosquito) [reply aiomisc/log/config.py:57]: Oops.

**Review by dizballanze — APPROVED** [review]

File: aiomisc/log/config.py (hunk 6 above)
Comment (mosquito) [reply aiomisc/log/config.py:15]: No the `Optional` is synonym to `Union[LogFormat, None]`, `None` raises an exception in this case.

File: aiomisc/log/enum.py (hunk 5 above)
Comment (mosquito) [reply aiomisc/log/enum.py:33]: [Here](https://github.com/python/cpython/blob/6bf3237379b17632db52cb39d181e8bac70173f3/Lib/logging/__init__.py#L589)

**Pull request merged**